src/composables/useBadgeData.js

```diff
@@ -1,26 +1,35 @@
-import { onMounted, ref, computed } from 'vue';
+import { ref, computed } from 'vue';
 import userAchievementProgressQuery from '#src/graphql/query/userAchievementProgress.graphql';
 import contentfulEntriesQuery from '#src/graphql/query/contentfulEntries.graphql';
 import logReadQueryError from '#src/util/logReadQueryError';
 
 /**
  * Utilities for loading and combining tiered badge data
  *
- * @param apollo The current Apollo client instance
- * @returns Cleaned up tiered badge data
+ * @returns Badge data and utilities
  */
-export default function useBadgeData(apollo) {
+export default function useBadgeData() {
 	const badgeAchievementData = ref();
 	const badgeContentfulData = ref();
 
+	/**
+	 * Gets a cleaned up version of Contentful badge data
+	 *
+	 * @param entry The Contentful entry
+	 * @returns The cleaned up Contentful badge data
+	 */
 	const getContentfulLevelData = entry => ({
 		id: entry?.fields?.key?.replace(/-level-\d+/, '') ?? '',
 		level: +(entry?.fields?.key?.replace(/\D/g, '') ?? ''),
 		levelName: entry?.fields?.challengeName ?? '',
+		challengeName: (entry?.fields?.challengeName ?? '').replace(/\s*✨\d+✨/, ''),
 		imageUrl: entry?.fields?.badgeImage?.fields?.file?.url ?? '',
 	});
 
-	const fetchAchievementData = () => {
+	/**
+	 * Calls Apollo to get the badge achievement service data
+	 */
+	const fetchAchievementData = apollo => {
 		apollo.query({ query: userAchievementProgressQuery })
 			.then(result => {
 				badgeAchievementData.value = result.data?.userAchievementProgress?.tieredLendingAchievements ?? [];
```

Comment:
Can we add `lendingAchievements` here?

Reply:
Wouldn't we need to use both to get all achievements (old and new)?

Reply:
We need both, for the journey and earned badges, these last ones includes both.
Probably we can also just call the new field `achievements` on the query, to get both in a call.

Reply:
What do you mean by the new field `achievements`? I'm not seeing that in the gql sandbox.

Reply:
For real? That one includes both cases.

<img width="497" alt="Screenshot 2024-10-28 at 5 08 25 p m" src="https://github.com/user-attachments/assets/b75c1d0e-391a-4c40-8bf4-0a70f7441dc5">

Reply:
Ah, on the parent, got it, thanks.

I'll do this in a follow-up PR.

Reply:
Thanks!